### PR TITLE
DOC-378: Add text for changes in CERT-3967.

### PR DESCRIPTION
**Pull request by steve-certora**
This PR documents the changes made in [CERT-3967](https://certora.atlassian.net/browse/CERT-3967).

Preserved blocks in invariants now support having the contract name as part of the method specification.  The block may have a `method` name, a `contract.method` name, or `*.method` (not counting the `fallback` form of the preserved block):

- If the block only has a `method` name, the preserved block only applies to method matches in the main contract.
- If the block has `contract.method` name, the preserved block only applies to method matches in the named contract.
- If the block has `*.method` name, the preserved block applies to all method matches in all contracts in the scene.

Note that if a preserved block with a `method` name exists and a preserved block with a more general name exists, such as `*.method`, the more specific preserved block takes precedence.

Before requesting review:
 - Make sure CI is passing
   - If the CI Details link gives a 404, you need to log in to readthedocs.com
 - Add link to generated documentation here
   - you can find this by following the read the docs link from the CI check
 - Ask for help in #documentation

Jira ticket: [DOC-378](https://certora.atlassian.net/browse/DOC-378)
Link to generated documentation: [here](https://certora-certora-prover-documentation--195.com.readthedocs.build/en/195/)

[CERT-3967]: https://certora.atlassian.net/browse/CERT-3967?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOC-378]: https://certora.atlassian.net/browse/DOC-378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/cvl/invariants.md [hunk 1]
```diff
@@ -37,8 +37,11 @@ preserved_block ::= "preserved"
                     [ "with" "(" params ")" ]
                     block
 
-method_signature ::= id "(" [ evm_type [ id ] { "," evm_type [ id ] } ] ")"
+method_signature ::= [ contract_name "." ] id  "(" [ evm_type [ id ] { "," evm_type [ id ] } ] ")"
                      | "fallback" "(" ")"
+
+contract_name ::= id
+                | "*"
```
Comment (naftali-g) [inline docs/cvl/invariants.md:44]: isn't this supposed to be `_`?

File: docs/cvl/invariants.md [hunk 2]
```diff
@@ -152,11 +155,20 @@ consists of the keyword `preserved` followed by an optional method signature,
...
-`preserved` block for all of them.
+The method signature of the preserved block may optionally contain a contract
+name followed by a `.` character followed by a contract method name.
+
+- In the case where the preserved block does not have a contract name but does
+have a method name (not the `fallback` case), the preserved block will apply
+only to methods that match in the main contract.
+- If the method signature includes a specific contract name, then the Prover
+only applies the preserved block to the methods in the named contract.
+- If the contract name is the wildcard character `*`, the Prover applies the
```
Comment (naftali-g) [inline docs/cvl/invariants.md:166]: same

File: docs/cvl/invariants.md [hunk 3]
```diff
@@ -152,11 +155,20 @@ consists of the keyword `preserved` followed by an optional method signature,
...
+preserved block to all contracts in the scene.
+
+If an invariant has multiple preserved blocks with the same method signature
+where one signature is more specific and the other is more general (as in 
+the `*.method` case), then the more specific preserved block will apply.
```
Comment (naftali-g) [inline docs/cvl/invariants.md:171]: same

**Review by naftali-g — DISMISSED** [review]

File: docs/cvl/invariants.md (hunk 1 above)
Comment (steve-certora) [reply docs/cvl/invariants.md:44]: After looking at the code, you are correct; it should be `_`.  For some reason, I was under the impression that the character was `*`.  Fixed.

File: docs/cvl/invariants.md (hunk 3 above)
Comment (steve-certora) [reply docs/cvl/invariants.md:171]: Fixed

**Review by shellygr — APPROVED** [review]
🐑

**Pull request merged**